integration-tests/src/tests/network/runner.rs

```diff
@@ -100,13 +101,13 @@ fn setup_network_node(
         runtime.store().clone(),
         client_config.chunk_request_retry_period,
     );
-    shards_manager_adapter.set_recipient(shards_manager_actor);
+    shards_manager_adapter.bind(shards_manager_actor);
     let peer_manager = PeerManagerActor::spawn(
         time::Clock::real(),
         db.clone(),
         config,
         Arc::new(near_client::adapter::Adapter::new(client_actor, view_client_actor)),
-        shards_manager_adapter,
+        shards_manager_adapter.as_sender(),
```

Comment:
while we are at it, why shards_manager_adapter is not a third argument of the near_client::adapter::Adapter ?  The idea was to make PeerManager as business logic agnostic as possible (i.e. it should be treated more like an RPC server boilerplate that you should plug RPC a set of handlers into, without the distinction of who does what.

Reply:
It's not clear to me what Adapter is intended to be - I thought it was just a way to avoid circular dependency from near-network to near-client. But either way, I was thinking that once we migrate Client and ViewClient to the Sender framework, then the "adapter" passed into the PeerManagerActor would just be a struct of `Sender<whatever the network is supposed to send to the Client>`, `AsyncSender<whatever the network is supposed to send to the view client>`, and `Sender<ShardsManagerRequestFromNetwork>`. Would that align with your mental model?